src/steamship/agents/llms/openai.py

```diff
@@ -38,3 +38,52 @@ def complete(self, prompt: str, stop: Optional[str] = None) -> List[Block]:
         action_task = self.generator.generate(text=prompt, options=options)
         action_task.wait()
         return action_task.output.blocks
+
+
+# TODO(dougreid): should this be a an extension of a BaseOpenAI or similar? Or completely separate?
+class ChatOpenAI(ConversationalLLM):
+    """LLM that uses Steamship's OpenAI plugin to generate completions.
+
+    NOTE: By default, this LLM uses the `gpt-3.5-turbo` model. Valid model
+    choices are `gpt-3.5-turbo` and `gpt-4`.
+    """
+
+    generator: PluginInstance
+    client: Steamship
+
+    def __init__(
+        self, client, model_name: str = "gpt-4-0613", temperature: float = 0.4, *args, **kwargs
+    ):
+        """Create a new instance.
+
+        Valid model names are:
+         - gpt-4
+         - gpt-3.5-turbo
+        """
+        client = client
+        generator = client.use_plugin(
+            PLUGIN_HANDLE, config={"model": model_name, "temperature": temperature}
+        )
+        super().__init__(client=client, generator=generator, *args, **kwargs)
+
+    def complete(self, prompt: str, stop: Optional[str] = None) -> List[Block]:
+        options = {}
+        if stop:
+            options["stop"] = stop
+        action_task = self.generator.generate(text=prompt, options=options)
```

Comment:
generate_task?  (Or does the returned task represent a tool action?)

Reply:
fwiw, this bit was copy/pasted from the other OpenAI LLM. `action_task` is somewhat appropriate for the Agent, as the generation is really action selection. we could call it `tool_selection_task` or similar to clear it up?